http/test/node/mock-http-source.ts

```diff
@@ -1,63 +1,23 @@
 import 'mocha';
 import * as assert from 'assert';
-import { mockHTTPSource, RequestOptions } from '../../src/index';
+import {mockHTTPDriver, RequestOptions} from '../../src/index';
 import xs from 'xstream';
-
-describe('mockHttpSource', function () {
-  it('should be in accessible in the API', function () {
-    assert.strictEqual(typeof mockHTTPSource, 'function');
-  });
-
-  it('should make an empty Observable stream of http responses on category foo', function (done) {
-    const userEvents = mockHTTPSource({
-      'foo': xs.empty(),
-    });
-    userEvents.select('foo').subscribe({
-      next: () => done(new Error('Should not emit any value')),
-      error: (err: any) => done(err),
-      complete: () => done(),
-    });
-  });
-
-  it('supports categorized streams', (done) => {
-    const response: any = {
-      text: 'Hello world',
+import {setup} from '../../../run/src';
+
+describe.only('mockHTTPDriver', function() {
```

Comment:
Most definitely. This latest commit doesn't even compile yet 😸 

I need a way to turn a `Partial<Response>` into something that implement the `Response` interface. Ideally I want to match superagent's behaviour perfectly, but most of the logic I need to utilize is tightly coupled with performing side effects.

In the past I've used https://github.com/M6Web/superagent-mock, so hopefully I can reuse some of the work they've done.